mmv1/third_party/terraform/services/spanner/resource_spanner_instance_config_test.go

```diff
@@ -0,0 +1,96 @@
+package spanner_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+// Acceptance Tests
+
+func TestAccSpannerInstanceConfig_basic(t *testing.T) {
+	t.Parallel()
+
+	idName := fmt.Sprintf("custom-nam11-config-%s", acctest.RandString(t, 10))
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckSpannerInstanceConfigDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSpannerInstanceConfig_basic(idName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("google_spanner_instance_config.basic", "replicas.0.location"),
+				),
+			},
+			{
+				ResourceName:      "google_spanner_instance_config.basic",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccSpannerInstanceConfig_update(t *testing.T) {
+	// Randomness
+	acctest.SkipIfVcr(t)
+	t.Parallel()
+
+	dName1 := fmt.Sprintf("custom-nam11-config-%s", acctest.RandString(t, 10))
+	dName2 := fmt.Sprintf("custom-nam11-config-%s", acctest.RandString(t, 10))
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckSpannerInstanceConfigDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSpannerInstanceConfig_update(dName1, dName1),
```

Comment:
the variable names here are a little confusing... the display name also doesn't need a random string in it. You could do something like:

```suggestion
				Config: testAccSpannerInstanceConfig_update(name, "display name"),
```